### PR TITLE
Added SDL_ShouldInit() and SDL_ShouldQuit()

### PR DESCRIPTION
**Pull request by slouken**
These are handy functions to support thread-safe initialization and shutdown.

### CODE REVIEW

File: include/SDL3/SDL_mutex.h [hunk 1]
```diff
@@ -30,6 +30,7 @@
 
 #include <SDL3/SDL_stdinc.h>
 #include <SDL3/SDL_error.h>
+#include <SDL3/SDL_thread.h>
```
Comment (Sackzement) [inline include/SDL3/SDL_mutex.h:33]: Should probably also include `SDL_atomic.h`, now that `SDL_AtomicInt` is used.

**Comment by Sackzement** [comment]
Would these changes in `SDL_mutex.h` fit better in the `SDL_thread.h` header?
All used types are already included there.

File: include/SDL3/SDL_mutex.h (hunk 1 above)
Comment (slouken) [reply include/SDL3/SDL_mutex.h:33]: SDL_thread.h already includes SDL_atomic.h, but it's probably better to be explicit.

**Comment by icculus** [comment]
Yeah, this should be in SDL_mutex.h

**Pull request merged**